assets/blocks/reader-revenue-manager/common/ButtonEdit.js

```diff
@@ -76,7 +76,10 @@ export default function ButtonEdit( {
 				).hasModuleAccess( 'reader-revenue-manager' );
 			}
 
-			setHasModuleAccess( hasModuleOwnershipOrAccess );
+			// Note: `hasModuleOwnershipOrAccess` can be expected to be undefined if
+			// - `ownerID` is not set for view-only or non-SK users.
+			// - the `check-access` endpoint returns a 403 error for non-SK users.
```

Comment:
Hey @nfmohit, apologies as I did suggest this comment, but on reading it back I've realised it could be a bit misleading, as we are not planning to show this component to non-SK users (as per https://github.com/google/site-kit-wp/issues/10400).

I'd therefore suggest this amendment for accuracy:

```suggestion
			// Note: `hasModuleOwnershipOrAccess` can be expected to be `undefined` if `ownerID` is not set for a view-only user.
```

The request to `check-access` won't be made for a view-only user, because it's guarded by a test for `hasModuleOwnershipOrAccess === false`, whereas `hasModuleOwnershipOrAccess` will always be `undefined` at that point because `ownerID` is not a view-only setting and thus always currently `undefined` in this context.

This does lead to question of whether `ownerID` _should_ be made a view-only setting, but we haven't done this so far for other modules which also make use of `hasModuleOwnershipOrAccess()`, so there's no need to consider that here.

I'll go ahead and apply this change myself to save the round trip, and merge the PR.